server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPServerTest.java

```diff
@@ -231,6 +232,98 @@ void serverShouldKeepReceivingErrorResponseWhenAnytime() throws Exception {
                 .isInstanceOf(SMTPConnectionClosedException.class)
                 .hasMessageContaining("421");
         }
+
+        @Test
+        void serverShouldDecreaseNumberOfAnswerAfterMatched() throws Exception {
+            MockSMTPBehavior behavior = new MockSMTPBehavior(
+                MAIL_FROM,
+                Optional.empty(),
+                Response.serverReject(SERVICE_NOT_AVAILABLE_421, "mock response"),
+                MockSMTPBehavior.NumberOfAnswersPolicy.times(5));
+
+            mockServer.setBehaviors(behavior);
+
+            sendMessageIgnoreError();
+
+            assertThat(mockServer.getBehaviorRepository()
+                    .remainingAnswers(behavior))
+                .contains(4);
+        }
+
+        @Test
+        void serverShouldRemoveBehaviorAfterGettingEnoughMatches() throws Exception {
```

Comment:
IMO no, this should just be a "no longer applied" behavior.

send your 5 messages, ignore error

Then send a message and assert it went ok: your behavior is no longer applied.